### PR TITLE
feat(outdated): don't fail when no lockfile & no deps

### PR DESCRIPTION
**Pull request by yaodingyd**
Related #2029

### CODE REVIEW

**Comment by zkochan** [comment]
Sorry, I am now on a business trip in India. I'll be slower in my reviews and replies.

**Comment by zkochan** [comment]
I think an easier and better solution for this problem would be making `bail=false` the default for the `outdated` command.

Also, the issue mentions that this fails when a package has no dependencies at all. If that's the case, that is a bug. Projects without dependencies are never outdated and they have to be just skipped

**Comment by yaodingyd** [comment]
I moved the lockfile checking logic after checking if project has dependencies

File: packages/outdated/test/outdated.spec.ts [hunk 1]
```diff
@@ -495,3 +495,58 @@ test('a dependency is not outdated if it is newer than the latest version', asyn
   t.deepEqual(outdatedPkgs, [])
   t.end()
 })
+
+test('outdated() should [] when there is no dependency', async (t) => {
+  const lockfile = {
```
Comment (zkochan) [inline packages/outdated/test/outdated.spec.ts]: this variable is not used in the test

**Comment by zkochan** [comment]
some tests are failing in `plugin-commands-outdated`

**Pull request merged**